### PR TITLE
Move BIP ENV to application-local

### PR DESCRIPTION
**Pull request by au2**
## What was the problem?

1. BIP environment variables for local development  and testing are not secrets anymore so they can be moved to application-local.properties from setenv.sh.
2. The https Rest template without server verification is not being used anymore. So it needs to be remove.
3. Few code cleaning changes needed in the mock services.

Associated tickets or Slack threads:

- [MCP-2364](https://amida.atlassian.net/browse/MCP-2364)

## How does this fix it?[^1]

1. BIP environment variables removed from setenv.sj and put in application-local.properties . I had left them to share with mock services but test and lint action is not calling setenv.sh so sharing is not possible. Instead of changing the action I am moving it back.
2. The https Rest template without server verification removed.
3. Code cleaning.

## How to test this PR
- Step 1
- Step 2[^secrel]

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: app/src/test/resources/application-test.yml [hunk 1]
```diff
@@ -37,20 +37,4 @@ lhAPIProvider:
   vroAudURL: "${VRO_AUD_URL:placeholderUrl}"
   apiKey: "${LH_VRO_API_KEY:placeholderKey}"
   validateToken: "NO"
-bip:
-  claimClientId: "${BIP_CLAIM_USERID:claimuser}"
```
Comment (wlin2019) [inline app/src/test/resources/application-test.yml:41]: Are these still required, aren't they?

**Review by wlin2019 — COMMENTED** [review]
It looks like end2endtest fail to get those ENV values.

File: app/src/main/resources/application-local.yml [hunk 2]
```diff
@@ -17,3 +17,19 @@ logging:
     org.hibernate.SQL: INFO
     org.type.descriptor.sql.BasicBinder: INFO
     org.hibernate.type.descriptor.sql.BasicBinder: INFO
+bip:
+  claimClientId: "${BIP_CLAIM_USERID:VRO_USER}"
+  claimSecret: "${BIP_CLAIM_SECRET:theSecret}"
```
[the remark this answers is not in the document]
Comment (au2) [reply app/src/main/resources/application-local.yml:22]: Yes, they are moved to application-local

File: app/src/test/resources/application-test.yml (hunk 1 above)
Comment (au2) [reply app/src/test/resources/application-test.yml:41]: they they are moved to application-local. This ones inherits that one.

Comment (wlin2019) [reply app/src/test/resources/application-test.yml:41]: Ok. I see.

File: app/src/main/resources/application-local.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (wlin2019) [reply app/src/main/resources/application-local.yml:22]: I see them. Why did I ask this question? It's my bad. I probably was looking at application-test when I typed the question.

**Pull request merged**